en/concept/flight_tasks.md

```diff
@@ -2,8 +2,141 @@
+
+     bool update();
+     bool activate(vehicle_local_position_setpoint_s last_setpoint);
+
+   private:
+     float _origin_z = 0.0f;
+   };
+   ```
+4. Update the cpp file as appropriate.
+   This example provides as simple implementation of **FlightTaskMyTask.cpp** that simply indicates that the task methods are called.
+   ```cpp
+   #include "FlightTaskMyTask.hpp"
+
+   bool FlightTaskMyTask::activate(vehicle_local_position_setpoint_s last_setpoint)
+   {
+     bool ret = FlightTask::activate(last_setpoint);
+     PX4_INFO("FlightTaskMyTask activate was called! ret:%d", ret); // check if engaged this mode.
+     return ret;
+   }
+
+   bool FlightTaskMyTask::update()
+   {
+     PX4_INFO("FlightTaskMyTask update was called!"); // check updated.
+     return true;
+   }
+   ```
+5. Add the new task to the list of tasks to be build in [/src/lib/flight_tasks/tasks/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/flight_tasks/tasks/CMakeLists.txt):  
+   ```
+   # add core flight tasks to list
+   list(APPEND flight_tasks_all
+     ManualAltitude
+     ManualAltitudeSmooth
+     ManualAltitudeSmoothVel
+     ManualPosition
+     ManualPositionSmooth
+     ManualPositionSmoothVel
+     AutoLineSmoothVel
+     AutoFollowMe
+     Offboard
+     Failsafe
+     Descend
+     Transition
+     MyTask
+     ${flight_tasks_to_add}
+   )
+   ```
+6. Update a flight mode to ensure that the task is called. 
+   Usually a parameter is used to select when a particular flight task should be used.
+   
+   For example, to enable our new `MyTask` in the Manual-Position control sub-mode:
+   - Update `MPC_POS_MODE` ([mc_pos_control_params.c](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/mc_pos_control/mc_pos_control_params.c)) to add an option for selecting "MyTask" if the parameter==2:
+     ```
+     ...
+     * @value 2 MyTask
+     * @value 3 Smooth position control (Jerk optimized)
+     * @group Multicopter Position Control
+     */
+     PARAM_DEFINE_INT32(MPC_POS_MODE, 3);
+     ``` 
+   - Add a switch in [MulticopterPositionControl.cpp](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/mc_pos_control/MulticopterPositionControl.cpp) to enable the task when `MPC_POS_MODE==2`"
+     ```cpp
+     ...
+     // manual position control
+     ...
+     switch (_param_mpc_pos_mode.get()) {
+       case 1:
+          error =  _flight_tasks.switchTask(FlightTaskIndex::ManualPositionSmooth);
+          break;
+
+       case 2: //Add option for new task: MyTask
+          error =  _flight_tasks.switchTask(FlightTaskIndex::MyTask);
+          break;
+     ...
+     ```
+  
+
+##  Test New Flight Task
+
+To test the flight stack you need to run the vehicle with the task enabled.
+For the example above, this means setting the parameter `MPC_POS_MODE=2`, taking off, and switching the vehicle to [Position mode](../flight_modes/position_mc.md).
+This would normally be done on the simulator.
```

Comment:
If someone follow this docs and test a flight with real vehicle, the vehicle may be start falling after switching MyTask mode because no setpoint is set.
Maybe some caution should be written...?

Reply:
I've modified this to make the point that it won't fly. Good catch :-)